src/network/network_survey.cpp

```diff
@@ -0,0 +1,409 @@
+
+/**
+ * List of all the generic setting tables.
+ *
+ * There are a few tables that are special and not processed like the rest:
+ * - _currency_settings
+ * - _misc_settings
+ * - _company_settings
+ * - _win32_settings
+ * As such, they are not part of this list.
+ */
+static auto &GenericSettingTables()
+{
+	static const SettingTable _generic_setting_tables[] = {
+		_difficulty_settings,
+		_economy_settings,
+		_game_settings,
+		_gui_settings,
+		_linkgraph_settings,
+		_locale_settings,
+		_multimedia_settings,
+		_network_settings,
+		_news_display_settings,
+		_pathfinding_settings,
+		_script_settings,
+		_world_settings,
+	};
+	return _generic_setting_tables;
+}
+
+/**
+ * Convert a settings table to JSON.
+ *
+ * @param survey The JSON object.
+ * @param table The settings table to convert.
+ * @param object The object to get the settings from.
+ */
+static void SurveySettingsTable(nlohmann::json &survey, const SettingTable &table, void *object)
+{
+	char buf[512];
+	for (auto &desc : table) {
+		const SettingDesc *sd = GetSettingDesc(desc);
+		/* Skip any old settings we no longer save/load. */
+		if (!SlIsObjectCurrentlyValid(sd->save.version_from, sd->save.version_to)) continue;
+
+		auto name = sd->GetName();
+		sd->FormatValue(buf, lastof(buf), object);
+
+		survey[name] = buf;
+	}
+}
+
+/**
+ * Convert settings to JSON.
+ *
+ * @param survey The JSON object.
+ */
+static void SurveySettings(nlohmann::json &survey)
+{
+	SurveySettingsTable(survey, _misc_settings, nullptr);
+#if defined(_WIN32) && !defined(DEDICATED)
+	SurveySettingsTable(survey, _win32_settings, nullptr);
+#endif
+	for (auto &table : GenericSettingTables()) {
+		SurveySettingsTable(survey, table, &_settings_game);
+	}
+	SurveySettingsTable(survey, _currency_settings, &_custom_currency);
+	SurveySettingsTable(survey, _company_settings, &_settings_client.company);
+}
+
+/**
+ * Convert generic OpenTTD information to JSON.
+ *
+ * @param survey The JSON object.
+ */
+static void SurveyOpenTTD(nlohmann::json &survey)
+{
+	survey["version"] = std::string(_openttd_revision);
+	survey["newgrf_version"] = _openttd_newgrf_version;
+	survey["build_date"] = std::string(_openttd_build_date);
+	survey["bits"] =
+#ifdef POINTER_IS_64BIT
+			64
+#else
+			32
+#endif
+		;
+	survey["endian"] =
+#if (TTD_ENDIAN == TTD_LITTLE_ENDIAN)
+			"little"
+#else
+			"big"
+#endif
+		;
+	survey["dedicated"] =
+#ifdef DEDICATED
+			"yes"
+#else
+			"no"
+#endif
+		;
+}
+
+/**
+ * Convert generic game information to JSON.
+ *
+ * @param survey The JSON object.
+ */
+static void SurveyConfiguration(nlohmann::json &survey)
+{
+	survey["network"] = _networking ? (_network_server ? "server" : "client") : "no";
+	if (_current_language != nullptr) {
+		std::string language_basename(_current_language->file);
+		auto e = language_basename.rfind(PATHSEPCHAR);
+		if (e != std::string::npos) {
+			language_basename.erase(0, e + 1);
+		}
+
+		survey["language"] = language_basename;
```

Comment:
Can't you assign a `std::string_view`? That would save you from moving characters around in `language_basename`.